cli/command_repository_set_parameters_test.go

```diff
@@ -183,3 +184,48 @@ func (s *formatSpecificTestSuite) TestRepositorySetParametersUpgrade(t *testing.
 
 	env.RunAndExpectSuccess(t, "index", "epoch", "list")
 }
+
+func (s *formatSpecificTestSuite) TestRepositorySetParametersRequiredFeatures(t *testing.T) {
+	env := s.setupInMemoryRepo(t)
+
+	env.RunAndExpectSuccess(t, "repository", "status")
+	env.RunAndExpectSuccess(t, "repository", "set-parameters", "--add-required-feature", "no-such-feature")
+	env.RunAndExpectFailure(t, "repository", "status")
```

Comment:
nit; Can we also verify the failure message to ensure it is not failing for unexpected reasons ?

Reply:
there's currently no mechanism for that, I want to refactor RunAndExpectFailure() to always require validation in separate PR.